FtpClient/mainwindow.cpp

```diff
@@ -51,6 +59,67 @@ void MainWindow::slotError(QAbstractSocket::SocketError err)
     }
 }
 
+void MainWindow::onChangeDir()
+{
+    std::string buffer;
+
+    buffer.resize(2);
+
+    *reinterpret_cast<short*>(&buffer[0]) = 0;
+
+    buffer += "Change dir plz";
+
+    socket->write( buffer.c_str(), buffer.size() );
+}
+
+void MainWindow::onDownloadFile()
+{
+    std::string buffer;
+
+    buffer.resize(2);
+
+    *reinterpret_cast<short*>(&buffer[0]) = 1;
+
+    buffer += "Download file plz";
+
+    socket->write( buffer.c_str(), buffer.size() );
+}
+
+void MainWindow::onUploadFile()
+{
+    QFile readStream("Your_Dir");
+    std::string directoryToSend("Your_Dir");
+
+    if( !readStream.open( QIODevice::ReadOnly ) )
+    {
+        qDebug() << "Can`t open file";
+        return;
+    }
+
+    std::vector<char> buffer(packageSize);
+
+    buffer[0] = 2;
+    buffer[1] = 0;
+
+    *reinterpret_cast<long*>(&buffer[2]) = readStream.size();
+
+    char *dataPtr = &buffer[6]; // 1024
```

Comment:
check out how operations are encapsulated in other pull request, do that way as well.